### ISSUE TITLE
io.readLine() unusable because EOF doesn't return current line

### ISSUE THREAD

**Issue by shawnl** [ziglang/zig#1579]
this patch isn't working for me either:

```
diff --git a/std/io.zig b/std/io.zig
index c10fdcb..a4d6094 100644
--- a/std/io.zig
+++ b/std/io.zig
@@ -626,11 +626,20 @@ pub fn readLine(buf: []u8) !usize {
     var stream = &adapter.stream;
     var index: usize = 0;
     while (true) {
-        const byte = stream.readByte() catch return error.EndOfFile;
+        var byte: u8 = undefined;
+        if (stream.readByte()) |b| {
+            byte = b;
+        } else |_| {
+            if (index == 0) {
+                return error.EndOfFile;
+            } else {
+                return index;
+            }
+        }
         switch (byte) {
             '\r' => {
                 // trash the following \n
-                _ = stream.readByte() catch return error.EndOfFile;
+                _ = stream.readByte() catch undefined;
                 return index;
             },
             '\n' => return index,
```

**Comment by shawnl** [comment]
hmm, maybe this is normal. ruby's `gets` does the same thing.

**Comment by Hejsil** [comment]
Fixed in #1801
